http/vibe/http/server.d

```diff
@@ -800,29 +800,51 @@ final class HTTPServerRequest : HTTPRequest {
 		*/
 		TLSCertificateInformation clientCertificate;
 
-		/** The _path part of the URL.
+		private bool urlParsed;
+		private void parseURL() @safe {
+			auto url = URL.parse(requestURL);
+			_path = urlDecode(url.pathString);
+			_queryString = url.queryString;
+			_username = url.username;
+			_password = url.password;
+			urlParsed = true;
+		}
 
-			Remarks: This field is only set if HTTPServerOption.parseURL is set.
+		/** The _path part of the URL.
 		*/
-		string path;
+		ref string path() @safe {
```

Comment:
There's a bit of code that expects these variables to be mutable (hence `ref`), e.g.

```d
http/vibe/http/auth/basic_auth.d(100,4): Error: function vibe.http.server.HTTPServerRequest.username () is not callable using argument types (string)
http/vibe/http/auth/digest_auth.d(119,4): Error: function vibe.http.server.HTTPServerRequest.username () is not callable using argument types (string)
http/vibe/http/server.d(391,2): Error: function vibe.http.server.HTTPServerRequest.path () is not callable using argument types (string)
http/vibe/http/server.d(392,2): Error: function vibe.http.server.HTTPServerRequest.queryString () is not callable using argument types (string)
http/vibe/http/server.d(393,2): Error: function vibe.http.server.HTTPServerRequest.username () is not callable using argument types (string)
http/vibe/http/server.d(394,2): Error: function vibe.http.server.HTTPServerRequest.password () is not callable using argument types (string)
```